docs/course_materials/compound_data_types.html

```diff
@@ -189,32 +244,171 @@ <h1><b>
+                        my_list = ['a', 'b', 'c', 'd', 'e']
+                        my_list[1:3] # ['b', 'c']
+                        my_list[1:] # ['b', 'c', 'd', 'e']
+                        my_list[:3] # ['a', 'b', 'c']
+                        my_list[:] # ['a', 'b', 'c', 'd', 'e']
+                        my_list[::2] # ['a', 'c', 'e']
+                    </code></pre>
+                </section>
+                <section>
+                    <h4>Important:</h4>
+                    <ul>
+                        <li>The <b>start</b> element is inclusive, which means that it will be included in the sub-list</li>
+                        <li>The <b>end</b> element is exclusive, which means that it won't be included in the sub-list</li>
+                        <li>The <b>step</b> element equals to 1 by default and doesn't need to be written always</li>
+                    </ul>
+                    <h4>Example:</h4>
+                    <pre><code data-trim data-noescape class="hljs lang-python">
+                        my_list[1:3] # ['b', 'c']
+                        #      [start    ]end
+                        #   0 ,  1 ,  2 ,  3 ,  4
+                        # ['a', 'b', 'c', 'd', 'e']
+                    </code></pre>
+                </section>
+                <section>
+                    <h3>
+                        <b>Negative indices</b>
+                    </h3>
+                    <h4>All three values in the slicing syntax may be negative</h4>
+                    <ul>
+                        <li>start - Starts listing elements from the end of the list</li>
+                        <li>end - Lists elements up until the last number of elements</li>
+                        <li>step - Reverses the order of the elements in the list</li>
+                    </ul>
+                </section>
+                <section>
+                    <h3>
+                        <b>Examples</b>
+                    </h3>
+                    <pre><code data-trim data-noescape class="hljs lang-python">
+                        my_list = ['a', 'b', 'c', 'd', 'e']
+                        # The last element
+                        my_list[-1] # ['e']
+                        # The last two elements
+                        my_list[-2:] # ['d', 'e']
+                        # Everything except the last two elements
+                        my_list[:-2] # ['a', 'b', 'c']
+                        # The entire list reversed
+                        my_list[::-1] # ['e', 'd', 'c', 'b', 'a']
+                        # The first three elements but reversed
+                        my_list[2::-1] # ['c', 'b', 'a']
+                        # The last two elements but reversed
+                        my_list[:-3:-1] # ['e', 'd']
+                        # Everything except the last two elements but reversed
+                        my_list[-3::-1] # ['c', 'b', 'a']
```

Comment:
This one looks a bit busy.
If I were a student I was confused.
Maybe it worth reducing some example or splitting